libraries/stdlib/common/src/generated/_Collections.kt

```diff
@@ -1320,7 +1325,12 @@ public inline fun <T, R> Iterable<T>.map(transform: (T) -> R): List<R> {
  * and returns the result of the transform applied to the element.
  */
 public inline fun <T, R> Iterable<T>.mapIndexed(transform: (index: Int, T) -> R): List<R> {
-    return mapIndexedTo(ArrayList<R>(collectionSizeOrDefault(10)), transform)
+    val collectionSize = collectionSizeOrDefault(10)
+    return when (collectionSize) {
+        0 -> emptyList()
+        1 -> listOf(transform(0, iterator().next()))
+        else -> mapIndexedTo(ArrayList<R>(collectionSize), transform)
```

Comment:
Such change makes `transform` lambda body inlined twice into the call site. It can lead to a significant bytecode size growth and is undesirable.